TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -65,13 +74,18 @@ public IChannelConnectionCaptureSerializer createOffloader(String connectionId)
                     // Used to essentially wrap Future returned by Producer to CompletableFuture
                     CompletableFuture cf = new CompletableFuture<>();
                     log.debug("Sending Kafka producer record: {} for topic: {}", recordId, topicNameForTraffic);
+                    MDC.put("diagnosticId", recordId);
+                    metricsLogger.info("Sending message to Kafka");
+                    MDC.remove("diagnosticId");
```

Comment:
This is error prone from a human perspective and an exception-safe one.  It's also far less efficient than just putting it into the logging line.  "diagnosticId" is going to be used by this thread in potentially many (global) places, and you're not preserving the old value.  This is a bad pattern.  Globals are bad and that's what these are.